### ISSUE TITLE
Cilium agent fails to compile BPF program resulting in error Could not locate previously compiled BPF template

### ISSUE THREAD

**Issue by bsiravara** [cilium/cilium#8817]
```
WARN [2019-08-03T02:38:06.468844998Z] cilium.io/agent: "Regeneration of endpoint failed" bpfCompilation=0s bpfLoadProg=0s bpfWaitForELF="148.26µs" bpfWriteELF=0s buildDuration="741.257µs" containerID= datapathPolicyRevision=1062 desiredPolicyRevision=1062 endpointID=3498 error="Could not locate previously compiled BPF template" identity=4 ipv4=10.0.197.40 ipv6= k8sPodName=/ mapSync="2.94µs" policyCalculation="4.821µs" prepareBuild="239.485µs" proxyConfiguration="4.448µs" proxyPolicyCalculation="15.072µs" proxyWaitForAck=0s reason="one or more identities created or deleted" subsys=endpoint waitingForCTClean="3.01µs" waitingForLock="1.098µs"
ERROR [2019-08-03T02:38:06.468917932Z] cilium.io/agent: "endpoint regeneration failed" containerID= datapathPolicyRevision=1062 desiredPolicyRevision=1062 endpointID=3498 error="Could not locate previously compiled BPF template" identity=4 ipv4=10.0.197.40 ipv6= k8sPodName=/ subsys=endpoint
```

**Comment by ianvernon** [comment]
Based off of offline discussion, I will look into the following:

* Do we try to recompile after first compilation fails?
* Should be we doing this template compilation at bootstrap?
* How do we make the failure more obvious?
* Should we fatal if this is the case? Have some type of backoff?
* Is there a way to configure the compiler to use a certain fraction of available RAM?

**Comment by ianvernon** [comment]
Of note - this was for the health endpoint.

**Comment by tgraf** [comment]
Fixed via #9203
